apps/ui/Application.cpp

```diff
@@ -0,0 +1,385 @@
+/* Copyright (c) 2019, EPFL/Blue Brain Project
+ * All rights reserved. Do not distribute without permission.
+ *
+ * This file is part of Brayns <https://github.com/BlueBrain/Brayns>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "Application.h"
+
+#include <brayns/Brayns.h>
+#include <brayns/common/input/KeyboardHandler.h>
+#include <brayns/engine/Camera.h>
+#include <brayns/engine/Engine.h>
+#include <brayns/engine/FrameBuffer.h>
+#include <brayns/engine/Renderer.h>
+#include <brayns/engine/Scene.h>
+#include <brayns/manipulators/AbstractManipulator.h>
+#include <brayns/parameters/ParametersManager.h>
+
+#include <imgui/imconfig.h>
+#include <imgui/imgui.h>
+// NOTE: include samples after imconfig.h and imgui.h
+#include <imgui/examples/imgui_impl_glfw.h>
+#include <imgui/examples/imgui_impl_opengl3.h>
+
+#include <algorithm>
+#include <cassert>
+
+Application::Application(brayns::Brayns& brayns, GLFWwindow* window,
+                         const int width, const int height)
+    : m_brayns(brayns)
+    , m_window(window)
+    , m_width(width)
+    , m_height(height)
+{
+    initOpenGL();
+    initImGUI();
+
+    auto& keyHandler = m_brayns.getKeyboardHandler();
+    keyHandler.registerKeyboardShortcut(
+        'z', "Switch between depth and color buffers", [&]() {
+            m_frameBufferMode = m_frameBufferMode == FrameBufferMode::DEPTH
+                                    ? FrameBufferMode::COLOR
+                                    : FrameBufferMode::DEPTH;
+        });
+    keyHandler.registerKeyboardShortcut('h', "Toggle help window", [&]() {
+        m_displayHelp = !m_displayHelp;
+    });
+}
+
+Application::~Application()
+{
+    if (m_fbTexture)
+    {
+        glDeleteTextures(1, &m_fbTexture);
```

Comment:
You're deleting Application with an atexit callback. Calling glDeleteTextures at that time is more than late. This is not critical, but you should review how you're doing the cleanup because segfaults at exit always give a bad impression (at least it does to me).